app/src/main/java/com/example/mymovie/MainActivity.kt

```diff
@@ -3,20 +3,39 @@ package com.example.mymovie
 import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.ImageView
 import android.widget.TextView
+import androidx.core.content.ContextCompat.startActivity
+import com.example.mymovie.databinding.ActivityMainBinding
+import com.example.mymovie.databinding.CardViewBinding
+import com.example.mymovie.databinding.FragmentMoviesDetailsBinding
+import com.example.mymovie.databinding.FragmentMoviesListBinding
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(R.layout.activity_main), FragmentMoviesListClickListener, BackToListClickListener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-
-        val textView: TextView = findViewById(R.id.first_text_view)
-        textView.setOnClickListener { moveToNextScreen() }
+        if (savedInstanceState == null) {
+            addMovieList()
+        }
     }
-
-    private fun moveToNextScreen() {
-        val intent = Intent(this, MovieDetailsActivity::class.java)
-
-        startActivity(intent)
+    private fun addMovieList() {
+        supportFragmentManager.beginTransaction()
+                .replace(R.id.main_container, FragmentMoviesList())
+                .addToBackStack(null)
+                .commit()
     }
-}
\ No newline at end of file
+    override fun toCardViewClicked() {
+        supportFragmentManager.beginTransaction()
+                .replace(R.id.main_container, FragmentMoviesDetails())
+                .addToBackStack(null)
+                .commit()
+    }
+    override fun backToListClicked() {
```

Comment:
just onBackPressed()
as u already added fragment to back stack system will handle it